src/ATokenVaultFactory.sol

```diff
@@ -0,0 +1,231 @@
+
+    /// @notice Mapping from deployer to deployed vaults
+    mapping(address => address[]) internal _vaultsByDeployer;
+
+    /// @notice Mapping to check if a vault was deployed by this factory
+    mapping(address => bool) internal _isVaultDeployed;
+
+    /// @notice Proxy admin address for all deployed vaults
+    address public immutable PROXY_ADMIN;
+
+    /*//////////////////////////////////////////////////////////////
+                                STRUCTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Struct containing constructor parameters for vault deployment
+     */
+    struct VaultParams {
+        address underlying;
+        uint16 referralCode;
+        IPoolAddressesProvider poolAddressesProvider;
+        address owner;
+        uint256 initialFee;
+        string shareName;
+        string shareSymbol;
+        uint256 initialLockDeposit;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                               CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Constructor
+     * @param proxyAdmin The address that will be the admin of all deployed proxies
+     */
+    constructor(address proxyAdmin) {
+        require(proxyAdmin != address(0), "ZERO_ADDRESS_NOT_VALID");
+        PROXY_ADMIN = proxyAdmin;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            EXTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Deploy a new ATokenVault with the given parameters
+     * @param params All parameters needed for vault deployment and initialization
+     * @return vault The address of the deployed vault proxy
+     */
+    function deployVault(VaultParams memory params) public returns (address vault) {
+        require(params.underlying != address(0), "ZERO_ADDRESS_NOT_VALID");
+        require(address(params.poolAddressesProvider) != address(0), "ZERO_ADDRESS_NOT_VALID");
+        require(params.owner != address(0), "ZERO_ADDRESS_NOT_VALID");
+        require(params.initialLockDeposit > 0, "ZERO_INITIAL_LOCK_DEPOSIT");
+        require(bytes(params.shareName).length > 0, "EMPTY_SHARE_NAME");
+        require(bytes(params.shareSymbol).length > 0, "EMPTY_SHARE_SYMBOL");
+
+        // Transfer the initial lock deposit from caller to this contract
+        IERC20(params.underlying).transferFrom(
+            msg.sender,
+            address(this),
+            params.initialLockDeposit
+        );
+
+        address implementation = ATokenVaultImplDeploymentLib.deployVaultImpl(
+            params.underlying,
+            params.referralCode,
+            params.poolAddressesProvider
+        );
+
+        vault = address(new TransparentUpgradeableProxy(
+            implementation,
+            PROXY_ADMIN,
+            ""
+        ));
```

Comment:
so this uses the dependency which is oz 4.7 but here we can also use the latest transparent proxy in aave deployments ie 5.1
it accepts owner (instead of proxyAdmin) as the input such that it deploys the proxy admin contract in the constructor, configures its owner and keeps the proxy admin on the proxy immutable so it's slightly cheaper

note if we're to go this route:
we're using the proxy factory deployed on all supported networks (https://github.com/bgd-labs/aave-address-book/blob/main/src/MiscEthereum.sol#L24-L25) for this deployment, since address-book isnt updated in this repo, i would use proxy_factory and owner as inputs